### PR TITLE
feat: add basic persistence, fixes

### PR DESCRIPTION
**Pull request by kincaidoneil**
- Add basic persistence to config file in user directory
- Fix XRP issues: #16, #9; temporary fix for #15 
- Fix issue with balance not applied for incoming packets
- Resolves ETH issues: #7, bug loading persisted claims (other issues were on server)
- Refactor some test code

### CODE REVIEW

File: src/settlement/machinomy.ts [hunk 1]
```diff
@@ -75,36 +71,44 @@ export const setupEngine = async (
...
-  settlerType: SettlementEngineType.Machinomy
-  privateKey: string
-  address: string
+  readonly settlerType: SettlementEngineType.Machinomy
+  readonly privateKey: string
+  readonly address: string
 }
 
+/**
+ * Ensure that the given string is begins with given prefix
```
Comment (karzak) [inline src/settlement/machinomy.ts]: typo: string is begins -> string begins

File: src/settlement/machinomy.ts [hunk 2]
```diff
@@ -75,36 +71,44 @@ export const setupEngine = async (
...
+ * - Prefix the string if it doesn't already
+ */
+const prefixWith = (prefix: string, str: string) =>
```
Comment (rhuairahrighairidh) [inline src/settlement/machinomy.ts]: rename to `ensurePrefixedWith`?

File: src/index.ts [hunk 3]
```diff
@@ -1,108 +1,107 @@
...
+  }
+  /* tslint:disable-next-line:readonly-keyword TODO */
+  credentials: ReadyCredentials[]
+  /* tslint:disable-next-line:readonly-keyword TODO */
+  uplinks: ReadyUplinks[]
 }
 
 export const connect = async (ledgerEnv: LedgerEnv = LedgerEnv.Testnet) => {
-  let state: State = {
+  const [fd, config] = await loadConfig()
```
Comment (rhuairahrighairidh) [inline src/index.ts]: I think we shouldn't hard code the directory where data is stored.
Maybe add a `dataDir: string` arg to `connect` and `loadConfig`. Then set a default in `connect` to `home/.switch/`.

File: src/index.ts [hunk 4]
```diff
@@ -1,108 +1,107 @@
...
+      [SettlementEngineType.Machinomy]: await Machinomy.setupEngine(ledgerEnv),
+      [SettlementEngineType.XrpPaychan]: await XrpPaychan.setupEngine(ledgerEnv)
     },
     credentials: [],
     uplinks: []
   }
 
-  // TODO Add functionality to connect existing uplinks based on config
-  //      (unnecessary/backburner until persistence is added)
+  // TODO Handle error cases if the uplinks fail to connect
```
[the remark this answers is not in the document]
Comment (kincaidoneil) [reply src/index.ts]: It hangs indefinitely...

**Comment by kincaidoneil** [comment]
@rhuairahrighairidh I opened #23 and #24 to track the persistence improvements

**Pull request merged**